pkg/cmd/tknpac/create/repository.go

```diff
@@ -76,15 +76,18 @@ func repositoryCommand(run *params.Run, ioStreams *cli.IOStreams) *cobra.Command
 			}
 
 			var providerName string
-			_, installationNS, err := bootstrap.DetectPacInstallation(ctx, createOpts.pacNamespace, run)
+			installed, installationNS, err := bootstrap.DetectPacInstallation(ctx, createOpts.pacNamespace, run)
+			if !installed {
+				return fmt.Errorf("pipelines-as-code is not installed in the cluster")
+			}
 			if err != nil {
-				return err
+				// if we don't have access to the pipelines as code info configmap (ie acl, don't let you)
+				// we won't try to update it
+				return createOpts.generateTemplate(nil)
 			}
 
 			if pacInfo.IsGithubAppInstalled(ctx, run, installationNS) {
-				if strings.Contains(createOpts.Event.URL, "github") {
```

Comment:
I remember that this if condition added to cover scenario like 

GitHubApp is installed but event.URL is for GitLab or BitBucket then it won't proceed for Webhook creation part that's why added this condition to make sure GitHubApp is installed and event URL is GitHub then just return and don't go for Webhook creation

Reply:
this is pretty buggy check, what about the gitea/bitbucket and others that somehow has a github string into the url?

moreoever we don't have any unit tests for this so we may as well guess what's right or not...

Reply:
that's why there was a switch before 🙃

Reply:
and this calls as well will fail on GHE i think too

Reply:
How about adding check here https://github.com/openshift-pipelines/pipelines-as-code/blob/26a61ccc270d864523c01c21fcdd9127b212bfc3/pkg/cmd/tknpac/create/repository.go#L93

I mean if GithubApp is installed even though

Reply:
How about adding check here https://github.com/openshift-pipelines/pipelines-as-code/blob/26a61ccc270d864523c01c21fcdd9127b212bfc3/pkg/cmd/tknpac/create/repository.go#L93 based on the providerName

I mean If GitHubApp is installed but provider name is other then Github proceed for webhook creation if not just don't got for webhook creation

WDYT?

Reply:
so what is worse? not allowing user to generate pipelinerun as non cluster admin for github apps or don't have the flow continue for other providers that is much less used ?

you can still create webhook creation with the tkn pac switch directly....

Reply:
what about you fix it in a following pr :D ?

Reply:
i think at some point we can just do tkn pac bootstrap github-webhook/gitlab/bitbucket/github-app and not care about being too smart with one command for everything

Reply:
Yes we can do that UX refactoring and change commands but that will take its own time as PAC is GA now and if we change some command we need to provide backward compatibility also

Reply:
> this is pretty buggy check, what about the gitea/bitbucket and others that somehow has a github string into the url?

> moreover we don't have any unit tests for this so we may as well guess what's right or not...

generally other SCM should not have github in there URL even if user provides that intentionally then its there responsibility i would say